### PR TITLE
feat: retrieve all rewards works 

### PR DESCRIPTION
**Pull request by Tulun**
**Motivation**

Voter should be able to collect all rewards available to them after round ends.

**Summary**

User may click the "collect" button on the wallet section if rewards are available; this will trigger a request to their wallet.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [ ]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [x]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: src/web3/post/retrieveRewards.ts [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+) => {
+  try {
+    const tx = await contract.functions[
+      "retrieveRewards(address,uint256,(bytes32,uint256,bytes)[])"
+    ](data.voterAddress, data.roundId, data.pendingRequests);
+
+    console.log("retrieve rewards TX?", tx);
+    return tx;
+  } catch (err) {
+    console.log("Err in attempted retrieveRewards transaction", err);
```
Comment (daywiss) [inline src/web3/post/retrieveRewards.ts:40]: some logs left in here. might want to just throw error

File: src/features/wallet/helpers/collectRewards.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    pendingRequestData.push(pendingRequest);
+  });
+
+  postData.pendingRequests = pendingRequestData;
+  retrieveRewards(contract, postData).then((tx) => {
+    tx.wait(1).then((conf: any) => {
+      // This function should collect all available rewards. Set balance to 0.
+      // Note: This still needs to be tested for N-1, N-2, ... rounds.
+      setAvailableRewards(DEFAULT_BALANCE);
+    });
```
Comment (daywiss) [inline src/features/wallet/helpers/collectRewards.ts:35]: just a nit here, but you can return the promise and unwind the nesting. also the call should be returned or caught if not returned
```js
  // this should be returned so we can catch error upstream
  return retrieveRewards(contract, postData)
     .then((tx) => tx.wait(1))
     .then((conf: any) => {
        // This function should collect all available rewards. Set balance to 0.
        // Note: This still needs to be tested for N-1, N-2, ... rounds.
        setAvailableRewards(DEFAULT_BALANCE);
     })
```

File: src/features/wallet/helpers/checkAvailableRewards.ts [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+) {
+  const promises = data.map(async (vote) => {
+    const rewardAvailable = await queryRetrieveRewards(
+      contract,
+      address,
+      vote.roundId,
+      vote.identifier,
+      vote.time
+    );
+    if (rewardAvailable) return parseFloat(rewardAvailable);
```
Comment (daywiss) [inline src/features/wallet/helpers/checkAvailableRewards.ts:18]: hopefully this doesnt affect reward getting, and only display, but this can cause some information loss / dust if value is used for claiming (which it doesnt look like it does).

File: src/features/wallet/helpers/checkAvailableRewards.ts [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+  });
+
+  return Promise.all(promises).then((values) => {
+    let balance = 0;
+    values.map((val) => {
+      if (val && val) return (balance += val);
```
Comment (daywiss) [inline src/features/wallet/helpers/checkAvailableRewards.ts:24]: is `val && val` a typo?

File: src/features/wallet/helpers/checkAvailableRewards.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      return false;
+    });
```
Comment (daywiss) [inline src/features/wallet/helpers/checkAvailableRewards.ts:26]: im surprised typescript is ok with this, returning bool or number.  just a nit but id probably recommend doing this as a reduce operation
```js
const values = await Promise.all(promises)
return values.reduce((balance,val)=>{
  if (val && val) return balance + val;
  return balance
},0)

```

File: src/features/wallet/helpers/checkAvailableRewards.ts [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+      contract,
+      address,
+      vote.roundId,
+      vote.identifier,
+      vote.time
+    );
+    if (rewardAvailable) return parseFloat(rewardAvailable);
+  });
+
+  return Promise.all(promises).then((values) => {
```
Comment (daywiss) [inline src/features/wallet/helpers/checkAvailableRewards.ts:21]: why not use await

File: src/features/wallet/helpers/calculateUMATotalValue.ts [hunk 7]
```diff
@@ -0,0 +1,9 @@
+export default function calculateUMATotalValue(price: number, balance: string) {
```
Comment (daywiss) [inline src/features/wallet/helpers/calculateUMATotalValue.ts:1]: this is cool, but just fyi ethers does have this utility [commify](https://docs.ethers.io/v5/api/utils/display-logic/#utils-commify), although your implementation is nice, so this is just an fyi

**Review by daywiss — APPROVED** [review]
looks good to me, thanks for keeping this pr small. did not really get to test the reveal stuff, but the wallet stats looked good.

File: src/features/wallet/helpers/collectRewards.ts (hunk 2 above)
Comment (Tulun) [reply src/features/wallet/helpers/collectRewards.ts:35]: Good call, that's cleaner. I think we may want to do some pending state stuff, but that's not currently happening.

File: src/features/wallet/helpers/checkAvailableRewards.ts (hunk 3 above)
Comment (Tulun) [reply src/features/wallet/helpers/checkAvailableRewards.ts:18]: Yeah this is just for the balance in the 3rd column. It should line up for actual values available, but it could be mismatched. But it won't matter -- the actual blockchain call will determine that.

File: src/features/wallet/helpers/checkAvailableRewards.ts (hunk 4 above)
Comment (Tulun) [reply src/features/wallet/helpers/checkAvailableRewards.ts:24]: Yeah I was trying to do promise.allSettled and forgot to change out.

File: src/features/wallet/helpers/calculateUMATotalValue.ts (hunk 7 above)
Comment (Tulun) [reply src/features/wallet/helpers/calculateUMATotalValue.ts:1]: Good to know, figured ethers didn't have something like that.

File: src/features/wallet/helpers/checkAvailableRewards.ts (hunk 5 above)
Comment (Tulun) [reply src/features/wallet/helpers/checkAvailableRewards.ts:26]: I refactored it to this, a balance can be undefined (promise):

  return values.reduce((balance, val) => {
    if (val) return balance ? balance : 0 + val;
    return balance;
  }, 0);

File: src/features/vote/ActiveRequestsForm.tsx [hunk 8]
```diff
@@ -199,7 +233,7 @@ const ActiveRequestsForm: FC<Props> = ({
         <thead>
           <tr>
             <th>Requested Vote</th>
-            <th>Proposal Detail</th>
+            {/* <th>Proposal Detail</th> */}
```
Comment (mrice32) [inline src/features/vote/ActiveRequestsForm.tsx:236]: Why is this commented out?

If it's meant to stay commented, might be useful to add a note explaining why.

File: src/features/vote/ActiveRequestsForm.tsx [hunk 9]
```diff
@@ -213,7 +247,7 @@ const ActiveRequestsForm: FC<Props> = ({
                 <td>
                   <div className="identifier">{el.identifier}</div>
                 </td>
-                <td>{el.ancillaryData}</td>
+                {/* <td>{el.ancillaryData}</td> */}
```
Comment (mrice32) [inline src/features/vote/ActiveRequestsForm.tsx:250]: Same as above.

File: src/features/vote/ActiveRequestsForm.tsx [hunk 10]
```diff
@@ -61,12 +67,30 @@ const ActiveRequestsForm: FC<Props> = ({
...
   const { data: roundId } = useCurrentRoundId();
   const { isOpen, open, close, modalRef } = useModal();
+  const [canReveal, setCanReveal] = useState(false);
+
+  useEffect(() => {
+    if (encryptedVotes.length && votePhase === "Reveal") {
+      if (revealedVotes.length) {
+        revealedVotes.forEach((el) => {
+          const findRevealedVote = encryptedVotes.find(
+            (x) => x.identifier === el.identifier
```
Comment (mrice32) [inline src/features/vote/ActiveRequestsForm.tsx:82]: Will this cause false-positives (and thus cause set reveal to be false when it should be true)? What happens if there are multiple votes in a round with the same identifier, but different ancillary data and/or timestamp?

Don't we need to compare ancillary data and timestamp too?

File: src/features/vote/ActiveRequestsForm.tsx [hunk 11]
```diff
@@ -183,11 +208,20 @@ const ActiveRequestsForm: FC<Props> = ({
         } else {
           datum.vote = vote;
         }
+        const findReveal = revealedVotes.find(
+          (x) => x.identifier === el.identifier
+        );
```
Comment (mrice32) [inline src/features/vote/ActiveRequestsForm.tsx:213]: Same question as above.

File: src/features/vote/ActiveRequestsForm.tsx [hunk 12]
```diff
@@ -20,6 +20,7 @@ import { formatVoteDataToCommit } from "./helpers/formatVoteDataToCommit";
 import { EncryptedVote } from "web3/get/queryEncryptedVotesEvents";
 
 import { FormWrapper, ModalWrapper } from "./styled/ActiveRequestsForm.styled";
+import { VoteRevealed } from "web3/get/queryVotesRevealedEvents";
```
Comment (mrice32) [inline src/features/vote/ActiveRequestsForm.tsx:23]: nit: are any of the changes in this file related to the retrieve rewards feature? This isn't a show-stopper on this PR, but in the interest of reducing PR sizes, maybe changes like this should be put into a separate PR in the future?

Thoughts? Or am I misunderstanding the purpose of the changes in this file?

File: src/features/vote/ActiveRequestsForm.tsx [hunk 13]
```diff
@@ -245,15 +279,20 @@ const ActiveRequestsForm: FC<Props> = ({
...
+                      ? "Revealed"
+                      : votePhase === "Reveal" &&
+                        el.vote !== UNDEFINED_VOTE &&
+                        !el.revealed
                       ? "Reveal"
-                      : /* this is incorrect -- just wanna hide this string for now. */
-                        null}
+                      : votePhase === "Reveal" && el.vote === UNDEFINED_VOTE
+                      ? "Uncommitted"
+                      : null}
```
Comment (mrice32) [inline src/features/vote/ActiveRequestsForm.tsx:295]: Thoughts on an alternative approach where you do the following above the JSX:
```js
tableValues.map((el, index) => {
  let stateEntry;
  let voteEntry;
  switch(votePhase) {
    case "Commit":
      // Logic
      break;
    case "Reaveal":
      // Logic
      break;
  }
}
// .. return JSX down here
```

There's certainly a tradeoff here -- the logic is placed away from where it's displayed. I only raise the point because nested ternaries are pretty hard to follow, but there doesn't seem to be a good alternative when the logic is inside the JSX.

**Review by mrice32 — COMMENTED** [review]
Looks awesome! Some initial comments. Will continue the review shortly!

File: src/features/vote/ActiveRequestsForm.tsx (hunk 8 above)
Comment (Tulun) [reply src/features/vote/ActiveRequestsForm.tsx:236]: Oh yes, sorry, this is because we are hiding the anc data column for now. I can add a comment np, or just delete it. It's a trivial thing to add back in.

File: src/features/vote/ActiveRequestsForm.tsx (hunk 9 above)
Comment (Tulun) [reply src/features/vote/ActiveRequestsForm.tsx:250]: Same as above ;)

File: src/features/vote/ActiveRequestsForm.tsx (hunk 10 above)
Comment (Tulun) [reply src/features/vote/ActiveRequestsForm.tsx:82]: I think the data filters based on the current round, but I could verify that with a bigger data set. If so, you would just need to add those comparisons like you said.

File: src/features/vote/ActiveRequestsForm.tsx (hunk 12 above)
Comment (Tulun) [reply src/features/vote/ActiveRequestsForm.tsx:23]: Oh this was largely finishing off some work that was tagged onto my larger previous PR. as it was a rather small addition I didn't feel the need to do a PR separate

Highly supportive of smaller PRs, though, we just need a fast turn around.

File: src/features/vote/ActiveRequestsForm.tsx (hunk 13 above)
Comment (Tulun) [reply src/features/vote/ActiveRequestsForm.tsx:295]: Yeah it's more a stylistic choice.

To be honest, because of how complex this ActiveRequest form view *is* in general, it is tempting to split off the commit and reveal states into their own views in general. There are potentially enough differences between the views on my chatting with Tom that this is probably okay. To be honest, the "reveal" stage, this is actually no longer a form, not in the sense we'd think about in an HTML sense.

File: src/features/vote/ActiveRequestsForm.tsx (hunk 8 above)
Comment (mrice32) [reply src/features/vote/ActiveRequestsForm.tsx:237]: Up to you -- I think either an explanatory note or deleting SGTM!

File: src/features/vote/ActiveRequestsForm.tsx (hunk 12 above)
Comment (mrice32) [reply src/features/vote/ActiveRequestsForm.tsx:23]: Yeah, I would argue that this is the exact sort of change that warrants its own PR. I think smaller PRs often _help_ with fast turnaround bc reviewers tend to procrastinate on PRs that are 100+ lines in my experience.

File: src/features/vote/ActiveRequestsForm.tsx (hunk 10 above)
Comment (mrice32) [reply src/features/vote/ActiveRequestsForm.tsx:82]: Yep, I'm saying I think it needs to compare ancillaryData and timestamp in addition to the identifier. And yeah, agreed that roundId comparison shouldn't be necessary.

File: src/features/vote/ActiveRequestsForm.tsx (hunk 13 above)
Comment (mrice32) [reply src/features/vote/ActiveRequestsForm.tsx:296]: Agreed on all of the above. I'm not super strongly opinionated. My gut response to seeing nested ternaries is that it's hard to read. But I totally get the constraints of JSX. Just wanted to mention it, but seems like you've already given it a decent amount of thought.

File: src/features/vote/ActiveRequestsForm.tsx (hunk 10 above)
Comment (Tulun) [reply src/features/vote/ActiveRequestsForm.tsx:82]: Sounds good! I'll add it in.

File: src/features/wallet/helpers/calculateUMATotalValue.ts (hunk 7 above)
Comment (mrice32) [reply src/features/wallet/helpers/calculateUMATotalValue.ts:1]: Any reason not to use that instead of the regex as a matter of simplicity?

File: src/features/wallet/helpers/collectRewards.ts [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+  PostRetrieveReward,
+  PendingRequestRetrieveReward,
+} from "web3/post/retrieveRewards";
+
+const DEFAULT_BALANCE = "0";
+
+export default function collectRewards(
+  contract: ethers.Contract,
+  data: RewardsRetrieved[],
+  setAvailableRewards: Function
```
Comment (mrice32) [inline src/features/wallet/helpers/collectRewards.ts:14]: Can this type be more specific?

File: src/features/wallet/helpers/checkAvailableRewards.ts [hunk 15]
```diff
@@ -0,0 +1,27 @@
...
+    );
+    if (rewardAvailable) return parseFloat(rewardAvailable);
+  });
+
+  const values = await Promise.all(promises);
+
+  return values.reduce((balance, val) => {
+    if (val) return (balance ? balance : 0) + val;
+    return balance;
+  }, 0);
```
Comment (mrice32) [inline src/features/wallet/helpers/checkAvailableRewards.ts:26]: Can this be simplified or am I missing something? I assume you should be able to tell typescript that balance _must_ be defined, right?
```suggestion
  return values.reduce((balance, val) => balance + val ? val : 0, 0);
```

File: src/features/wallet/helpers/calculateUMATotalValue.ts (hunk 7 above)
Comment (Tulun) [reply src/features/wallet/helpers/calculateUMATotalValue.ts:5]: Nope! That works just fine. 

Fixed, with an improvement for the 0 balance case as well.

File: src/features/wallet/helpers/collectRewards.ts [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+  postData.pendingRequests = pendingRequestData;
+  // this should be returned so we can catch error upstream
+  return (
+    retrieveRewards(contract, postData)
+      // wait for at least 1 block conf.
+      .then((tx) => tx.wait(1))
+      .then((conf: any) => {
+        // This function should collect all available rewards. Set balance to 0.
+        // Note: This still needs to be tested for N-1, N-2, ... rounds.
+        setAvailableRewards(DEFAULT_BALANCE);
```
Comment (mrice32) [inline src/features/wallet/helpers/collectRewards.ts:38]: This shouldn't necessarily collect for _all_ rounds, right? In fact, if you haven't claimed for multiple rounds, it _can't_ collect all rewards, right?

File: src/features/wallet/helpers/formatWalletBalance.ts [hunk 17]
```diff
@@ -0,0 +1,28 @@
+// There are two different colours for the first 4 digits and last 4 digits of the number.
+// Hence we need to return an array of strings.
+// We want to limit the overall string length to ~8 digits.
+export default function formatWalletBalance(balance: string): string[] {
```
Comment (mrice32) [inline src/features/wallet/helpers/formatWalletBalance.ts:4]: Similar to my other comment above, I think this might be better suited for a different PR unless this deals with rewards (although, now that it's here, I think it's fine to keep in).

File: src/features/wallet/helpers/checkAvailableRewards.ts [hunk 18]
```diff
@@ -0,0 +1,27 @@
...
+import { VoteRevealed } from "web3/get/queryVotesRevealedEvents";
+import { queryRetrieveRewards } from "web3/get/queryRetrieveRewards";
+
+export default async function checkAvailableRewards(
+  data: VoteRevealed[],
+  address: string,
+  contract: ethers.Contract
+) {
+  const promises = data.map(async (vote) => {
+    const rewardAvailable = await queryRetrieveRewards(
```
Comment (mrice32) [inline src/features/wallet/helpers/checkAvailableRewards.ts:11]: Minor note (not sure if there's anything to change): it looks like we're casting this to a float then using it to determine _if_ there are any rewards to claim. If we have rewards that are so small that they are below the precision limit of JS floats, we may accidentally truncate nonzero rewards to 0 and think that the user has _no_ rewards to claim when they really have just a very small number. I'm not sure how small their initial balance would need to be for this to happen, but just wanted to mention it since this float seems to be used for more than just display.

**Review by mrice32 — COMMENTED** [review]
A few more comments, but claim logic generally looks quite clean!

File: src/features/wallet/helpers/collectRewards.ts (hunk 14 above)
Comment (Tulun) [reply src/features/wallet/helpers/collectRewards.ts:14]: setAvailableRewards you mean?

File: src/features/wallet/helpers/checkAvailableRewards.ts (hunk 18 above)
Comment (daywiss) [reply src/features/wallet/helpers/checkAvailableRewards.ts:11]: i noticed this as well, and you bring up a really good edge case. I think in general we should default to handling these numbers as wei strings/bn's until its time to display / math them. and maybe define some guidelines around how to deal with BNs across all our apps since its tricky.

File: src/web3/get/queryRetrieveRewards.ts [hunk 19]
```diff
@@ -1,22 +1,40 @@
...
+      {
+        identifier: stringToBytes32(identifier),
+        time: Number(time),
+        ancillaryData,
+      },
     ]);
 
     return ethers.utils.formatEther(reward.toString());
   } catch (err) {
+    console.log("identifier", identifier);
```
Comment (mrice32) [inline src/web3/get/queryRetrieveRewards.ts:36]: Is this console log meant to be left in here?

File: src/web3/createMulticallContractInstance.ts [hunk 20]
```diff
@@ -0,0 +1,35 @@
+import MulticallAtrifact from "@uma/core/build/contracts/MulticallMock.json";
```
Comment (mrice32) [inline src/web3/createMulticallContractInstance.ts:1]: nit: probably slightly preferable to import the interface `Multicall` rather than `MulticallMock` (the implementation) since you shouldn't need the full implementation to know how to interact with it.

File: src/web3/createMulticallContractInstance.ts [hunk 21]
```diff
@@ -0,0 +1,35 @@
...
+  // const artifact: Network = MulticallAtrifact.networks;
+  // const network = artifact[networkId];
+  // console.log("network", network, "artifact", artifact, MulticallAtrifact);
+
+  const contract = new ethers.Contract(
+    // network.address,
+    // This is stubbed in for now. We need to develop this formally for mainnet / kovan.
+    process.env.REACT_APP_CURRENT_ENV === "test"
+      ? "0xD6b61cAd80DBe2Fe26Fa672b8Ec2FdAf002CeBbC"
+      : "",
```
Comment (mrice32) [inline src/web3/createMulticallContractInstance.ts:29]: nit: you can pull in pre-deployed addresses by network id by either importing the makerdao file that lists them via their package:
```
import multicallAddresses from "@makerdao/multicall/src/addresses";

const networkName = "kovan";
const multicallAddress = multicallAddresses[networkName];
```

OR what might be easier is just to copy the [mapping](https://github.com/makerdao/multicall.js/blob/master/src/addresses.json) into this file:
```
{
  "mainnet": {
    "multicall": "0xeefba1e63905ef1d7acba5a8513c70307c1ce441",
    "rpcUrl": "https://mainnet.infura.io"
  },
  "kovan": {
    "multicall": "0x2cc8688c5f75e365aaeeb4ea8d6a480405a48d2a",
    "rpcUrl": "https://kovan.infura.io"
  },
  "rinkeby": {
    "multicall": "0x42ad527de7d4e9d9d011ac45b31d8551f8fe9821",
    "rpcUrl": "https://rinkeby.infura.io"
  },
  "goerli": {
    "multicall": "0x77dca2c955b15e9de4dbbcf1246b4b85b651e50e",
    "rpcUrl": "https://rpc.slock.it/goerli"
  },
  "xdai": {
    "multicall": "0xb5b692a88bdfc81ca69dcb1d924f59f0413a602a",
    "rpcUrl": "https://dai.poa.network"
  }
}
```

You could remove the RPC urls and change the names to network IDs to make it easier to look up at runtime.

File: src/hooks/useMulticall.ts [hunk 22]
```diff
@@ -0,0 +1,30 @@
...
+  ] = useState<ethers.Contract | null>(null);
+
+  useEffect(() => {
+    // If connected, try to create contract with assigned signer.
+    if (isConnected) {
+      // Signer can be null check for null and if we've already defined a contract.
+      if (signer && !multicallContract && network) {
+        const contract = createMulticallContractInstance(
+          signer
+          // network.chainId.toString()
```
Comment (mrice32) [inline src/hooks/useMulticall.ts:22]: Commented out?

File: src/web3/createMulticallContractInstance.ts [hunk 23]
```diff
@@ -0,0 +1,35 @@
...
+import { ethers } from "ethers";
+
+// interface Network {
+//   [key: string]: {
+//     address: string;
+//     events: object;
+//     links: object;
+//     transactionHash: string;
+//   };
+// }
```
Comment (mrice32) [inline src/web3/createMulticallContractInstance.ts:11]: Commented out?

File: src/features/wallet/helpers/collectRewards.ts [hunk 24]
```diff
@@ -0,0 +1,76 @@
...
+  setAvailableRewards: Dispatch<SetStateAction<string>>,
+  multicallContract: ethers.Contract
+) {
+  const multicallCollectRewards = [] as MulticallCollectRewards[];
+  const votingInterface = new ethers.utils.Interface(VotingArtifact.abi);
+
+  // find the unique roundIds
+  const uniqueRoundIds = data
+    .map((item) => item.roundId)
+    .filter((value, index, self) => self.indexOf(value) === index);
```
Comment (mrice32) [inline src/features/wallet/helpers/collectRewards.ts:32]: I really wish that the array had a native `.unique()` method rather than having to implement it yourself or use lodash, so this wouldn't be required to do something so simple.

File: src/features/wallet/helpers/collectRewards.ts [hunk 25]
```diff
@@ -0,0 +1,76 @@
...
+
+const DEFAULT_BALANCE = "0";
+
+interface MulticallCollectRewards {
+  // contract address
+  target: string;
+  callData: string;
+}
+
+export default function collectRewards(
```
Comment (mrice32) [inline src/features/wallet/helpers/collectRewards.ts:19]: +1 this function looks great!

File: src/features/wallet/helpers/collectRewards.ts [hunk 26]
```diff
@@ -0,0 +1,76 @@
...
+  PostRetrieveReward,
+  PendingRequestRetrieveReward,
+} from "web3/post/retrieveRewards";
+
+import VotingArtifact from "@uma/core/build/contracts/Voting.json";
+
+const DEFAULT_BALANCE = "0";
+
+interface MulticallCollectRewards {
+  // contract address
```
Comment (mrice32) [inline src/features/wallet/helpers/collectRewards.ts:14]: comment?

**Review by mrice32 — APPROVED** [review]
LGTM! Only a few minor comments around how we're pulling in multicall.

File: src/web3/createMulticallContractInstance.ts (hunk 21 above)
Comment (Tulun) [reply src/web3/createMulticallContractInstance.ts:29]: Good to know. If there are already kovan / mainnet deployments out there, we just need to worry about the locally deployed ones.

File: src/web3/createMulticallContractInstance.ts (hunk 23 above)
Comment (Tulun) [reply src/web3/createMulticallContractInstance.ts:11]: Only used if the networks object is defined. I wasn't sure if this would be the case, but it's looking like we aren't going to be using artifacts anyway.
